porcupine/plugins/anchors.py

```diff
@@ -0,0 +1,173 @@
+                anchor.startswith(self.custom_anchor_prefix)
+                and self.tab_textwidget.index(anchor) == cursor_index
+            ):
+                self.tab_textwidget.mark_unset(anchor)
+                self.linenumbers.do_update()
+                return
+        self.tab_textwidget.mark_set(
+            self.custom_anchor_prefix + str(time.time()), "insert linestart"
+        )
+
+        self.linenumbers.do_update()
+
+    def jump_to_next(self, event: tkinter.Event[tkinter.Misc]) -> None:
+        cursor_row = self._get_cursor_index().split(".")[0]
+        anchor_list = self._get_anchors()
+        anchor_rows = reversed(
+            sorted(
+                [
+                    int(self.tab_textwidget.index(anchorpoint).split(".")[0])
+                    for anchorpoint in anchor_list
+                ]
+            )
+        )
+
+        next_anchor_row = 0
+        for anchor_row in anchor_rows:
+            if anchor_row > int(cursor_row):
+                next_anchor_row = anchor_row
+            else:
+                break
+
+        if not next_anchor_row:
+            return
+        else:
+            self.tab_textwidget.mark_set("insert", f"{str(next_anchor_row)}.0")
+
+        # If cursor is below last row
+        if int(self._get_cursor_index().split(".")[0]) > int(
+            self.tab_textwidget.index(f"@0,{self.tab_textwidget.winfo_height()}").split(".")[0]
+        ):
+            self.tab_textwidget.see("insert")
+
+        # text.see("insert")
+
+        # first_line = int(self._textwidget.index("@0,0").split(".")[0])
+        # last_line = int(
+        #     self._textwidget.index(f"@0,{self._textwidget.winfo_height()}").split(".")[0]
+        # )
+
+        # TODO: If user jumps to the last anchor and then to next, it should jump to the first one in the file.
+
+    def jump_to_previous(self, event: tkinter.Event[tkinter.Misc]) -> None:
+        cursor_row = self._get_cursor_index().split(".")[0]
+        anchor_list = self._get_anchors()
+        anchor_rows = sorted(
+            [
+                int(self.tab_textwidget.index(anchorpoint).split(".")[0])
+                for anchorpoint in anchor_list
+            ]
+        )
+
+        next_anchor_row = 0
+        for anchor_row in anchor_rows:
+            if anchor_row < int(cursor_row):
+                next_anchor_row = anchor_row
+            else:
+                break
+
+        if not next_anchor_row:
+            return
+        else:
+            self.tab_textwidget.mark_set("insert", f"{str(next_anchor_row)}.0")
+
+        # If cursor is above first row
+        if int(self._get_cursor_index().split(".")[0]) < int(
+            self.tab_textwidget.index("@0,0").split(".")[0]
+        ):
+            self.tab_textwidget.see("insert")
+
+        # TODO: Current Bugs:
+        # 1) Jump to previous will delete code on current row if no more anchors above.
+        # 2) Can't commit/push because of git merge weirdness
```

Comment:
Looks like `2)` got resolved. Is `1)` still a problem? I don't see any code that should change the contents of the text widget.

Reply:
Comment 2 has also been deleted.

Yes, comment 1 is a bug. If you jump to the first anchor with "jump_to_previous", and then continue clicking ctrl+k, code from that row is deleted.

Reply:
This is the default behavior for pressing Ctrl+K. It happens even without the plugin. To fix it, change your methods to `return "break"`. This tells Tk that you have handled the event and nothing else should do anything about it.